rfcs/0024-Non-consensus-node-protocol.md

```diff
@@ -0,0 +1,224 @@
+||**$\epsilon =0.4$**|**$\epsilon =0.34$**|**$\epsilon =0.1$**(bitcoin)| **$b = 6 mins$**(current) | **$b = 2 mins$** | **$b = 1 min$** | **$b = 30 secs$** |
+| 5 | 0.49024 |0.386024|0.04501 |30m| 10m| 5m|2.5m|
+| 30| 0.313376 |0.127204|2.3263e-06 |3h| 1 | 30m |15m|
+| 50| 0.228895 |0.057254|1.04804e-09|5h| 1h 40m | 50m| 25m|
+| 80| 0.146897 |0.0179358|1.01212e-14 |8h | 2h 40m | 1h 20m|40m|
+| 100| 0.110262|0.00835778|4.59285e-18 |10h| 3h 20m |1h 40m |50m|
+| 200| 0.0273769 |0.00019081|8.84e-35 |20h | 6h 40m |3h 20m|1h 40m|
+| 300| 0.00697404 |4.43371e-06|1.70148e-51 |30h | 10h |5h|2h 30m|
+| 400| 0.00179144 |1.0336e-07|3.27493e-68 |40h | 13h 20m |6h 40m|3h 20m|
+| 500| 0.000461763 |2.4114e-09|6.30342e-85 |50h | 16h 40m |8h 20m|4h 10m|
+<!--| 600| 0.000119216 |5.62686e-11|1.21325e-101 |60h | 20h |10h|5h|
+| 700| 3.0803e-05 |1.31306e-12 |2.33521e-118 |70h | 23h 20m |11h 40m|5h 50m|
+| 800| 7.96218e-06 |3.06414e-14 |4.49469e-135 |80h |26h 40m |13m 20m|6h 40m|
+| 900| 2.05858e-06 | 8.00374e-16|8.65116e-152 |90h |30h |15h|7h 30m|
+| 1000| 5.32297e-07|1.66863e-17|1.66513e-168 |100h | 33h 20m |16h 40m|8h 20m|-->
+
+Whatever value of `v` we choose, the selection mechanism between the two blocks would still use the `k` defined for full nodes and therefore, the fork resolution rules will not be affected even in the case of forks of length greater than `v`.
+Also, the above probabilities are for forks that are caused by adversarial behaviors. There can be forks due to random network partitions and since `v` is going to very small compared to `k` we have to consider forks due to short network partitions and determine their frequency/length in some heuristic way.
+
+But within a parition, there is an assumption that all the nodes will resolve any fork of length 1 i.e., every node will be in sync on the best chain within `2*delta` slots if they hear about it. So `v` should be at least `2*delta` blocks (not using slots here to avoid cases when there are no blocks at all within this period since delta is small enough for that to be possible). The value of delta we want is ~4 therefore `v` >= 8 blocks. With the above probabilities for $\epsilon =0.34$ (which is the lowest Vanishree would like it to be), `v` = 50 seems good enough.
+
+### Gossip layer
+
+The gossip protocol for non-consensus nodes should be separated from the one for full nodes because messages like blocks, transactions, and snark works are not required and having a different gossip layer for messages pertaining to the non-consensus nodes will reduce the bandwidth usage significantly.
+
+    1. Layer 1: Existing gossip layer that includes broadcasting blocks and snark pool diffs.
+    2. Layer 2: New layer for gossiping information needed by non-consensus nodes- 
+        a) Block headers
+    3 Layer 3: For transactions that nodes can send. This is a separate layer because we don't want non-consensus nodes (or archive nodes/snark workers) to receive any.
```

Comment:
These names are bad. People are not going to remember which is which, and the numbering implies that layer n+1 is built on top of layer n, which really isn't the case here. We can have a different pubsub topic for each of full blocks, block headers, snark bundles, and user transactions, and just refer to them by name.

Also, for reliability's sake full nodes should participate in all the topics.